### PR TITLE
Fix incorrect check when attempting to boot from existing disk.

### PR DESCRIPTION
**Pull request by iosdevzone**
**TL;DR** This pull request fixes a problem where LisaEm will refuse to boot from an existing, valid DC42 ProFile disk chosen by the user at start up.

## The Details

When LisaEm is powered on without an existing disk, one of the options presented is **Select existing ProFile instead of creating a new.**

<p align="center">
LisaEmHardDriveSizeDialog
</p>

If this option is chosen and a valid ProFile DC42 image is chosen, LisaEm will complain. 

<p align="center">
SorryNotALisaHardDiskImage
</p>

This is caused by the following piece of code in [lisaem_wx.cpp](https://github.com/arcanebyte/lisaem/blob/6660f7b7eaea09f6b2c648cdd2a9bfa70dc7a8a3/src/host/wxui/lisaem_wx.cpp#L10080-L10088):
```c
check = dc42_auto_open(&P, filename, "");
// if we failed to open, or it's not at least a 5MB dc42 image, alert and don't accept
if ((!check) || P.datasizetotal < (9728 * 512))                                  {
    filename[0] = 0; // clear return file
    messagebox("The selected file is either not a proper DC42 image, or it's not a ProFile/Widget image.", "Sorry Not a Lisa Hard Disk Image");                   }
```
The problem is that `dc42_auto_open` returns `0` on success, so the correct error check is `check` not `!check`.

Deleting this one `!` character allows the hard disk to open without issues.

### CODE REVIEW

**Comment by arcanebyte** [comment]
Thank you for the PR, and sorry for the delay on this. Will check this out and merge if it looks good.
